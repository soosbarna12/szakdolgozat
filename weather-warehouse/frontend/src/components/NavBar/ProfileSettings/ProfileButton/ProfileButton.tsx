import PersonOutlineOutlinedIcon from '@mui/icons-material/PersonOutlineOutlined';
import { Box, Tooltip } from "@mui/material";
import React from "react";
import { StyledIconButton } from '../../../../stlyes/button.style';
import { ProfileMenu } from "../ProfileMenu/ProfileMenu";
import { ProfileButtonProps } from "./ProfileButton.type";


export function ProfileButton(props: Readonly<ProfileButtonProps>) {
  const { isLightTheme, handleSetLightTheme } = props;
  const [anchorElUser, setAnchorElUser] = React.useState<null | HTMLElement>(null);

  function handleOpenMenu(event: React.MouseEvent<HTMLElement>) {
    setAnchorElUser(event.currentTarget);
  }

  function handleCloseMenu() {
    setAnchorElUser(null);
  }

  return (
    <Box sx={{ flexGrow: 0 }} data-testid="profileButtonContainer">
      <Tooltip title="Open settings">
        <StyledIconButton
          onClick={handleOpenMenu}
          sx={{ boxShadow: 4 }}
          data-testid="profileButton">
          <PersonOutlineOutlinedIcon sx={{ padding: 0, margin: 0, width: "32px", height: "32px" }} />
        </StyledIconButton>
      </Tooltip>
      <ProfileMenu
        isLightTheme={isLightTheme}
        handleSetLightTheme={handleSetLightTheme}
        handleCloseMenu={handleCloseMenu}
        anchorElUser={anchorElUser}
        data-testid="profileMenu"/>
    </Box>
  );
}
