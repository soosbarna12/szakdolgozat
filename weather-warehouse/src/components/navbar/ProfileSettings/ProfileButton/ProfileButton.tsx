import PersonOutlineOutlinedIcon from '@mui/icons-material/PersonOutlineOutlined';
import { Box, IconButton, Tooltip } from "@mui/material";
import React from "react";
import { ProfileMenu } from "../ProfileMenu/ProfileMenu";
import { ProfileButtonProps } from "./ProfileButton.type";


export function ProfileButton(props: Readonly<ProfileButtonProps>) {
  const { handleSetTheme } = props;
  const [anchorElUser, setAnchorElUser] = React.useState<null | HTMLElement>(
    null
  );
  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElUser(event.currentTarget);
  };

  const handleCloseUserMenu = () => {
    setAnchorElUser(null);
  };

  return (
    <Box sx={{ flexGrow: 0 }}>
      <Tooltip title="Open settings">
        <IconButton
          onClick={handleOpenUserMenu}
          color='secondary'
          sx={{ padding: 0, margin: "10px", width: "42px", height: "42px", boxShadow: 4, outline: 1 }}>
          <PersonOutlineOutlinedIcon sx={{ padding: 0, margin: 0, width: "32px", height: "32px" }} />
        </IconButton>
      </Tooltip>
      <ProfileMenu
        handleSetTheme={handleSetTheme}
        handleCloseUserMenu={handleCloseUserMenu}
        anchorElUser={anchorElUser} />
    </Box>
  );
}
