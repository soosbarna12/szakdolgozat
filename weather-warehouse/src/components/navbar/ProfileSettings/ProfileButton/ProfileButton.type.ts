export interface ProfileButtonProps {
	isLightTheme: boolean;
	handleSetLightTheme: (isLightTheme: boolean) => void;
}
