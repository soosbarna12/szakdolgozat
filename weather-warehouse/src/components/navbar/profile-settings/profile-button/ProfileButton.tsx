import { Box, IconButton, Tooltip } from "@mui/material";
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import React from "react";
import { ProfileMenu } from "../profile-menu/ProfileMenu";

export function ProfileButton() {
  const [anchorElUser, setAnchorElUser] = React.useState<null | HTMLElement>(null);

  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElUser(event.currentTarget);
  };  

  const handleCloseUserMenu = () => {
    setAnchorElUser(null);
  };

  return (
        <Box sx={{ flexGrow: 0 }}>
        <Tooltip title="Open settings">
          <IconButton onClick={handleOpenUserMenu} sx={{ p: 0 }}>
            <AccountCircleIcon/>
          </IconButton>
        </Tooltip>
        <ProfileMenu handleCloseUserMenu={handleCloseUserMenu} anchorElUser={anchorElUser}/>
      </Box>
      
    );
  }