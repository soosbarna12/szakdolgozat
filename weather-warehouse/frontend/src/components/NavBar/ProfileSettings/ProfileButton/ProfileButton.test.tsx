import '@testing-library/jest-dom';
import { fireEvent, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { renderWithQueryClient } from '../../../../utils/test/renderWithQueryClient';
import { ProfileButton } from './ProfileButton';

jest.mock('../ProfileMenu/ProfileMenu', () => ({
  ProfileMenu: ({

    handleSetLightTheme,
    handleCloseMenu,
    anchorElUser,
  }: {
    isLightTheme: boolean;
    handleSetLightTheme: () => void;
    handleCloseMenu: () => void;
    anchorElUser: HTMLElement | null;
  }) => (
    <div data-testid="profileMenu">
      {anchorElUser && (
        <div>
          <button data-testid="toggleThemeButton" onClick={handleSetLightTheme}>
            Toggle Theme
          </button>
          <button data-testid="closeMenuButton" onClick={handleCloseMenu}>
            Close Menu
          </button>
        </div>
      )}
    </div>
  ),
}));

describe('NavBar/ProfileSettings/ProfileButton', () => {
  const mockHandleSetLightTheme = jest.fn();

  const renderComponent = (isLightTheme = true) => {
    return renderWithQueryClient(
      <ProfileButton
        isLightTheme={isLightTheme}
        handleSetLightTheme={mockHandleSetLightTheme}
      />
    );
  };

  it('matches ProfileButton component snapshot', () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });

  it('renders the ProfileButton and opens the ProfileMenu on click', async () => {
    renderComponent();

    // Verify the button is rendered
    const profileButton = screen.getByRole('button', { name: /open settings/i });
    expect(profileButton).toBeInTheDocument();

    // Click the button to open the menu
    fireEvent.click(profileButton);

    // Verify the menu is open
    await waitFor(() => {
      expect(screen.getByTestId('profileMenu')).toBeInTheDocument();
    });
  });

  it('calls handleSetLightTheme when the toggle theme button is clicked', async () => {
    renderComponent();

    // Open the menu
    const profileButton = screen.getByRole('button', { name: /open settings/i });
    fireEvent.click(profileButton);

    // Click the toggle theme button
    const toggleThemeButton = await screen.findByTestId('toggleThemeButton');
    fireEvent.click(toggleThemeButton);

    // Verify the toggle theme function is called
    expect(mockHandleSetLightTheme).toHaveBeenCalled();
  });
});