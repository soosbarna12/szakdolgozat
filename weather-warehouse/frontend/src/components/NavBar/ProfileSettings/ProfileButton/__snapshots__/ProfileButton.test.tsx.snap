// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar/ProfileSettings/ProfileButton matches ProfileButton component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-2uchni"
        data-testid="profileButtonContainer"
      >
        <button
          aria-label="Open settings"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-7l956p-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="profileButton"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gtean2-MuiSvgIcon-root"
            data-testid="PersonOutlineOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4m0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4"
            />
          </svg>
        </button>
        <div
          data-testid="profileMenu"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-2uchni"
      data-testid="profileButtonContainer"
    >
      <button
        aria-label="Open settings"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-7l956p-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="profileButton"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gtean2-MuiSvgIcon-root"
          data-testid="PersonOutlineOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4m0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4"
          />
        </svg>
      </button>
      <div
        data-testid="profileMenu"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
